src/internal/storage/fileset/index/reader.go

```diff
@@ -52,28 +54,43 @@ func NewReader(chunks *chunk.Storage, cache *Cache, topIdx *Index, opts ...Optio
 	return r
 }
 
+// ctxWithMetrics returns a child context with relevant reader metrics.
+func ctxWithMetrics(ctx context.Context) context.Context {
```

Comment:
I think we should remove the skipped indexes meter because it is an unclear concept. It doesn't clearly correspond to a range or file type index, so I think it is just going to lead to confusion. The others seem good, although I would change `readFiles` to `emittedFiles`.

Reply:
I've broken it down into skipped files and skipped ranges. Keeping metrics for skipped ranges and files is important since that's what allowed us to confirm our initial theory for Batelle's recent support issue.

Reply:
The skipped ranges don't really tell us anything useful, and the skipped indexes didn't help in the Battelle case. What helped was knowing the total ranges / files traversed. The skipped ranges are only relevant for finding the starting point, which is logarithmic regardless of how the client uses the index reader. Skipped files could potentially be helpful with respect to the datum filtering functionality, but that functionality probably shouldn't exist in the first place. I'm fine with leaving both for now, but, like most information we collect, I think they have more noise than signal.

Reply:
You're right. I checked the message history and it was traversed files that was over 2 million. Based on that, I'll drop the skipped ranges for now since you mentioned that skipped files may be helpful for datum filtering